azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/auth/AzureClientFactory.java

```diff
@@ -29,9 +26,7 @@
     public static final String SUBSCRIPTION_NOT_SPECIFIED = "Subscription ID was not specified, using the first subscription in current account," +
             " please refer https://github.com/microsoft/azure-maven-plugins/wiki/Authentication#subscription for more information";
 
-    public static Azure getAzureClient(AuthType authType, AuthConfiguration auth, String subscriptionId) throws IOException, AzureLoginFailureException {
-        final AzureEnvironment environment = AzureEnvironment.AZURE;
-        final AzureTokenWrapper azureTokenCredentials = authType.getAzureToken(auth, environment);
+    public static Azure getAzureClient(AzureTokenWrapper azureTokenCredentials, String subscriptionId) throws IOException, AzureLoginFailureException {
         if (azureTokenCredentials != null) {
```

Comment:
throw NPE for null azureTokenCredentials

Reply:
Using `Preconditions.checkNotNull` to do the validation